File: backend/src/main/java/br/net/mirante/colaborador/domain/dtos/ClienteDTO.java
package br.net.mirante.colaborador.domain.dtos;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ClienteDTO implements BaseDTO {

    private Long id;

    @NotBlank
    @Size(min = 3, max = 100)
    private String nome;

    @NotBlank
    @Size(max = 11)
    private String cpf;

    private EnderecoDTO endereco;
    private EmailDTO email;
    private TelefoneDTO telefone;

}

File: frontend/src/main/angular/src/app/shared/shared-models/dto/cliente-dto.ts
import {EmailDto} from "./email-dto";
import {TelefoneDto} from "./telefone-dto";
import {EnderecoDto} from "./endereco-dto";

export interface ClienteDto {
  id: number;
  nome?: string;
  cpf?: string;
  endereco?: EnderecoDto;
  email?: EmailDto;
  telefone?: TelefoneDto;
}
